### DISCUSSION TITLE
Error Trapping

### DISCUSSION THREAD

**Discussion by smartinsight-at** [AlexaCRM/dynamics-webapi-toolkit#55] [Q&A] [answered]
I am just getting started with this toolkit.

I am trapping errors using the following. This seems to work.

                    try
                    {
                    $contactId = $client->Create( $contact );
                    }
                    catch (Exception $e) {
                        echo 'ATTENTION!<br><br>'.$e->getMessage().'<br><br>';
                        exit;
                    }

Is there a better way?

**Comment by georged** [comment, marked as answer]
@smartinsight-at 

catching exceptions is the standard way of trapping the error. What do you do with it is totally up to you. If you have a logger wired up then the error will appear in the log as well. For example:

```php
use Monolog\Logger;
use Monolog\Handler\StreamHandler; 

$log = new Logger( 'whatever' );
$log->pushHandler(new StreamHandler('php://stdout', Logger::INFO));

$settings = new \AlexaCRM\WebAPI\OData\OnlineSettings();
$settings->setLogger( $log );
```

This will generate error information and dump it into stdout. Usually handlers created for logging into files, etc.

HTH
